src/services/class.service.ts

```diff
@@ -56,40 +56,41 @@ export class ClassService {
   }
 
   public getSimilarClasses(
-    message: types.IMessage,
+    interaction: discord.CommandInteraction,
     invalidClasses: string[],
     action: 'register' | 'unregister'
-  ): types.IEmbedData[] {
+  ): types.IInteractionEmbedData[] {
     return invalidClasses.map((invalidClass: string) => {
-      const emojiData: types.IEmojiTable[] = [];
+      const buttonTable: types.IButtonTable[] = [];
       const embeddedMessage: discord.MessageEmbed = new discord.MessageEmbed();
 
       embeddedMessage.setColor('#0099ff').setTitle(`${invalidClass} Not Found`);
       const shouldShowAuthorOnRegister = true;
       if (shouldShowAuthorOnRegister) {
-        embeddedMessage.setAuthor(this._messageService.getEmbedAuthorData(message));
+        embeddedMessage.setAuthor(this._messageService.getEmbedAuthorData(interaction));
       }
 
       const [similarClassID] = this.findSimilarClasses(invalidClass);
 
       // TODO check if similarity is close then decide whether to return the guess or tell them to get a mod.
 
-      embeddedMessage.setDescription(
-        `Did you mean \`${similarClassID}\`?\n` +
-          `React with ✅ to ${action} for this class.\n` +
-          'React with ❎ to close this offering.'
-      );
+      embeddedMessage.setDescription(`Did you mean \`${similarClassID}\`?\n`);
```

Comment:
Unfortunately yes, autocomplete doesn't force you to choose an option it just suggests, people can still put in any arbitrary values. It probably won't be likely they do this, but this flow is still here just in case that happens.